src/meshio/netgen/_netgen.py

```diff
@@ -0,0 +1,371 @@
+        cells[-1][1].append(pi)
+        cells_index[-1].append(index)
+        if skip_every_second_line:
+            line, is_eof = _fast_forward_over_blank_lines(f)
+
+
+def _write_cells(f, block, index=None):
+    if len(block) == 0:
+        return
+    pmap = np.array(meshio_to_netgen_pmap[block.type])
+    dim = _topological_dimension[block.type]
+    post_data = []
+    pre_data = []
+    i_index = 0
+    if dim == 0:
+        post_data = [1]
+        i_index = 1
+    elif dim == 1:
+        pre_data = [1, 0]
+        post_data = [-1, -1, 0, 0, 1, 0, 1, 0]
+    elif dim == 2:
+        pre_data = [1, 1, 0, 0, len(pmap)]
+        i_index = 1
+    elif dim == 3:
+        pre_data = [1, len(pmap)]
+
+    col1 = len(pre_data)
+    col2 = col1 + len(pmap)
+    col3 = col2 + len(post_data)
+
+    pi = np.zeros((len(block), col3), dtype=np.int32)
+    pi[:, :col1] = np.repeat([pre_data], len(block), axis=0)
+    pi[:, col1:col2] = block.data[:, pmap] + 1
+    pi[:, col2:] = np.repeat([post_data], len(block), axis=0)
+    if index is not None:
+        pi[:, i_index] = index
+    np.savetxt(f, pi, "%i")
+
+
+def _skip_block(f):
+    n = int(f.readline())
+    for _ in range(n):
+        f.readline()
+
+
+def read_buffer(f):
+    points = []
+    cells = []
+    cells_index = []
+
+    have_edgesegmentsgi2_in_two_lines = False
+
+    # check if "mesh3d" is the first non-empty, non-comment line
+    while True:
+        line, is_eof = _fast_forward_over_blank_lines(f)
+        if line == "":
+            continue
+        elif line == "mesh3d":
+            break
+        else:
+            raise RuntimeError("Not a valid Netgen mesh")
+
+    while True:
+        line, is_eof = _fast_forward_over_blank_lines(f)
+        if is_eof:
+            break
+
+        elif line == "dimension":
+            dimension = int(f.readline())
+        elif line == "geomtype":
+            geomtype = int(f.readline())
+            if geomtype not in [0, 1, 10, 11, 12, 13]:
+                logging.warning(f"Unkown geomtype in Netgen mesh: {geomtype}")
+
+        elif line == "points":
+            npoints = int(f.readline())
+            if npoints > 0:
+                points = np.loadtxt(f, max_rows=npoints)
+                if dimension == 2:
+                    points = points[:, :2]
+
+        elif line in [
+            "pointelements",
+            "edgesegments",
+            "edgesegmentsgi",
+            "edgesegmentsgi2",
+            "surfaceelements",
+            "surfaceelementsgi",
+            "surfaceelementsuv",
+            "volumeelements",
+        ]:
+            _read_cells(f, line, cells, cells_index, have_edgesegmentsgi2_in_two_lines)
+        elif line == "endmesh":
+            break
+
+        elif line == "identificationtypes":
+            _ = int(f.readline())  # num_identifiactions
+            _ = f.readline()  # identifications in one line
+
+        elif line == "surf1   surf2      p1      p2":
```

Comment:
Is the number of spaces important? Otherwise, one could do line.spli() == [..., ...].